packages/typespec-ts/src/index.ts

```diff
@@ -286,7 +286,11 @@ export async function $onEmit(context: EmitContext) {
           dpgContext,
           modularEmitterOptions,
           "api",
-          subClient
+          subClient,
```

Comment:
it seems we set `hierarchyClient` as true at the very beginning: https://github.com/azure/autorest.typescript/blob/main/packages/typespec-ts/src/transform/transfromRLCOptions.ts#L442.